### PR TITLE
Expose constructor erasure in reflection interface

### PR DESCRIPTION
**Pull request by cmcmA20**
Currently there's no way to distinguish erased constructors from normal ones. I couldn't find the corresponding issue, please link it with this PR.

Continuation of (abandoned) #7247 but with a limited scope.

### CODE REVIEW

**Comment by cmcmA20** [comment]
@andreasabel , is there anything else to take care of?

**Review by andreasabel — APPROVED** [review]
Thanks!  Maybe someone reflection-active should review as well.

**Review by jespercockx — APPROVED** [review]

**Comment by jespercockx** [comment]
It seems there's something that needs to be updated in the standard library. The standard process for this is to *first* make a new branch of the standard library compatible with this PR, then point this PR to that branch of the standard library, merge this PR, and finally create a PR to the experimental branch of the stdlib with the change.

**Comment by cmcmA20** [comment]
> The standard process for this is to _first_ make a new branch of the standard library compatible with this PR, then point this PR to that branch of the standard library, merge this PR, and finally create a PR to the experimental branch of the stdlib with the change.

Sorry, can you please elaborate on this process? Do you mean I need to branch off the current agda-stdlib master? Then how to point this PR to the new branch?

**Comment by jespercockx** [comment]
Sorry, yes. This really ought to be documented somewhere. The process is as follows:
1. Locally create a new branch of the standard library, called `issue7322-compat` (or something similar)
2. Push this branch to somewhere (the official stdlib repo or your own fork of it).
3. Create a PR from this branch to the `experimental` branch (https://github.com/agda/agda-stdlib/tree/experimental)
4. Edit the `.gitmodules` file in your local copy of the agda repository to point to this branch of the stdlib (also update the url if you used your own fork), and push it to the branch of this PR.
5. Wait until this PR is merged.
6. Once this PR is merged, update your branch of the stdlib from step 2 so it uses the latest commit on agda/master
7. Finally, the PR to the std-lib experimental branch is merged

**Comment by jespercockx** [comment]
Something strange is going on here and I'm not sure what. It seems like just updating the `.gitmodules` file is not enough, you also need to update the submodule itself. However, whatever I do Github does not like it and shows that there are conflicts that need to be resolved. I'm a bit rusty on how to work with submodules myself, perhaps someone else has an idea of how to do it properly. (Also, sorry for the mess with the sequence of force-pushes, I was trying to make it work but failed.)

**Comment by jespercockx** [comment]
@andreasabel do you happen to know what is going wrong here with the submodule?

**Comment by andreasabel** [comment]
I rebased it on `master`, seems to have resolved the conflict.

**Comment by andreasabel** [comment]
@jespercockx wrote:
> Sorry, yes. This really ought to be documented somewhere. The process is as follows:
> 
>     1. Locally create a new branch of the standard library, called `issue7322-compat` (or something similar)
> 
>     2. Push this branch to somewhere (the official stdlib repo or your own fork of it).
> 
>     3. Create a PR from this branch to the `experimental` branch ([agda/agda-stdlib@`experimental`](https://github.com/agda/agda-stdlib/tree/experimental?rgh-link-date=2024-06-28T14%3A18%3A08Z))
> 
>     4. Edit the `.gitmodules` file in your local copy of the agda repository to point to this branch of the stdlib (also update the url if you used your own fork), and push it to the branch of this PR.
> 
>     5. Wait until this PR is merged.
> 
>     6. Once this PR is merged, update your branch of the stdlib from step 2 so it uses the latest commit on agda/master
> 
>     7. Finally, the PR to the std-lib experimental branch is merged

But then, at the end, Agda master's `std-lib` submodule will no longer point to `experimental`.  So is there a step 8 missing that fixes this?

Update: Ad 3.: the PR to the std-lib needs to set the Agda commit hash in the CI to the new version of Agda that contains the feature to be merged.  Otherwise CI fails there.  This means the PR to agda should already be opened, or at least there should be a branch on agda/agda.

**Comment by cmcmA20** [comment]
@jespercockx @andreasabel Thank you very much for finishing this PR and for shedding some light on the process of merging such changes!

**Pull request merged**
